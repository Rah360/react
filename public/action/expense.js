import uuid from 'uuid'
import database from '../firbase/firebase'
export const addExpense = ({
    description = "",
    note = "",
    amount = "",
    createdAt = "",
    id=""
} = {}) => ({
    type: "ADD",
    expense: {
        id,
        description,
        note,
        amount,
        createdAt
    }
})

export const addFunction=(expenseData={})=>{
    let {
        description = "",
        note = "",
        amount = "",
        createdAt = ""
    
    }=expenseData
   const expense= {
            description,
            note,
            amount,
            createdAt
            
    }
    return (dispatch)=>{
        database.ref("notes").push({
            expense
        }).then((ref)=>{
           
                dispatch(addExpense({ ...expense, id: ref.key }))
        }).catch((e)=>{console.log('didnt work =>>>',e);
        })
    }
}

export const editExpense = (id, update) => ({
    type: "EDIT",
    id,
    update
})



export const removeExpense = ({
    id = 0
} = {}) => ({
    type: "RMV",
    id
})

export const RemoveFromDatabase=({id})=>{
    console.log("id",id)
    return (dispatch) =>{
        console.log(`notes/${id}`);
        
        database.ref(`notes/${id}`).set(null).then(()=>{
            console.log("Removed from database")
            dispatch(removeExpense({id}))
        })
    }
}