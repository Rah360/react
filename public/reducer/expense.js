export default (state = [], action) => {
    switch (action.type) {
        case "ADD":
            return [...state, action.expense]
        case "RMV":
            return state.filter((id) => {
                return id.id !== action.id
            })
        case "EDIT":
            return state.map((x) => {
                if (x.id == action.id) {
                    return {
                        ...x,
                        ...action.update
                    }
                } else {
                    return x
                }
            })
        default:
            return state
    }
}