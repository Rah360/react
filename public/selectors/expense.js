export default (expense, {
    text,
    startDate,
    endDate
}) => {
    return expense.filter((x) => {
        const start = typeof startDate !=="number" || startDate <= x.createdAt
        const end = typeof endDate !== "number" || endDate >= x.createdAt
        const textMatch = x.description.includes(text)
       
        return start && end && textMatch

    })
}