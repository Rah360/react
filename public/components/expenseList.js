import React from 'react'
import { connect } from 'react-redux'
import ListItem from './ListItem'
import visibleExpense from "../selectors/expense";
import {setText} from '../action/filter'

const expenseList=(props)=>(
    <div>
        <input type="text" placeholder="search bills here!" onChange={(e)=>{
            props.dispatch(setText(e.target.value))
        }}/>
        {props.expense.map(x=>{return <ListItem key={x.id} {...x}/>})}
    </div>
)

const mapStateToProps=(state)=>{
    // console.log("uuu",state)
    return {
        expense: visibleExpense(state.expenses, state.filter),
        filter:state.filter
    }
}

export default connect(mapStateToProps)(expenseList)