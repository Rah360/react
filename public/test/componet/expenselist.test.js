import React from 'react'
import { shallow } from 'enzyme'
import expense from '../../fixutre/expenses'
import expenseList from '../../components/expenseList'

test("testing expense list",()=>{
    const wrapper=shallow(<expenseList {...expense[0]}/>)
    expect(wrapper).toMatchSnapshot()
})

test("testing simulation expense list", () => {
    const wrapper = shallow(<expenseList {...expense[0]} />)
    expect(wrapper).toMatchSnapshot()
})