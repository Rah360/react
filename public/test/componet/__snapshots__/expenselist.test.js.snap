// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing expense list 1`] = `
<expenseList
  amount={195}
  description="Gum"
  id="1"
  note=""
/>
`;

exports[`testing simulation expense list 1`] = `
<expenseList
  amount={195}
  description="Gum"
  id="1"
  note=""
/>
`;
